Traceback (most recent call last):
  File "/home/runner/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/miniconda3/envs/leapfrog-fault-models/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/runner/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/home/runner/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Import modules 
from fault_mesh.faults.leapfrog import LeapfrogMultiFault
import os
import numpy as np
import geopandas as gpd
------------------


[31m---------------------------------------------------------------------------[39m
[31mModuleNotFoundError[39m                       Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[1][39m[32m, line 2[39m
[32m      1[39m [38;5;66;03m# Import modules [39;00m
[32m----> [39m[32m2[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mfaults[39;00m[34;01m.[39;00m[34;01mleapfrog[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m LeapfrogMultiFault
[32m      3[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mos[39;00m
[32m      4[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mnumpy[39;00m[38;5;250m [39m[38;5;28;01mas[39;00m[38;5;250m [39m[34;01mnp[39;00m

[36mFile [39m[32m~/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/fault_mesh/faults/leapfrog.py:26[39m
[32m     24[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mutilities[39;00m[34;01m.[39;00m[34;01mcutting[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m cut_line_between_two_points, cut_line_at_point
[32m     25[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mutilities[39;00m[34;01m.[39;00m[34;01mgraph[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m connected_nodes, suggest_combined_name
[32m---> [39m[32m26[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mfaults[39;00m[34;01m.[39;00m[34;01mconnected[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m ConnectedFaultSystem
[32m     28[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mutilities[39;00m[34;01m.[39;00m[34;01mmeshing[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m triangulate_contours
[32m     29[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mutilities[39;00m[34;01m.[39;00m[34;01msplines[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m spline_fit_contours

[36mFile [39m[32m~/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/fault_mesh/faults/connected.py:25[39m
[32m     23[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mutilities[39;00m[34;01m.[39;00m[34;01mmeshing[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m get_strike_dip_from_normal, fit_plane_to_points, weighted_circular_mean, most_common_or_first, triangulate_contours
[32m     24[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mutilities[39;00m[34;01m.[39;00m[34;01msplines[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m spline_fit_contours
[32m---> [39m[32m25[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mfault_mesh[39;00m[34;01m.[39;00m[34;01mfaults[39;00m[34;01m.[39;00m[34;01mmesh[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m FaultMesh
[32m     28[39m [38;5;28;01mclass[39;00m[38;5;250m [39m[34;01mConnectedFaultSystem[39;00m:
[32m     29[39m [38;5;250m    [39m[33;03m"""_summary_[39;00m
[32m     30[39m [33;03m    """[39;00m

[36mFile [39m[32m~/miniconda3/envs/leapfrog-fault-models/lib/python3.12/site-packages/fault_mesh/faults/mesh.py:10[39m
[32m      8[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mshapely[39;00m[34;01m.[39;00m[34;01mgeometry[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m box
[32m      9[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mscipy[39;00m[34;01m.[39;00m[34;01mspatial[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m ConvexHull
[32m---> [39m[32m10[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mnumba[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m njit
[32m     12[39m [38;5;129m@njit[39m
[32m     13[39m [38;5;28;01mdef[39;00m[38;5;250m [39m[34mcross_3d[39m(a, b):
[32m     14[39m     [38;5;28;01mreturn[39;00m np.array([
[32m     15[39m         a[[32m1[39m]*b[[32m2[39m] - a[[32m2[39m]*b[[32m1[39m],
[32m     16[39m         a[[32m2[39m]*b[[32m0[39m] - a[[32m0[39m]*b[[32m2[39m],
[32m     17[39m         a[[32m0[39m]*b[[32m1[39m] - a[[32m1[39m]*b[[32m0[39m]
[32m     18[39m     ])

[31mModuleNotFoundError[39m: No module named 'numba'

